pkg/load/agents/configAgent.go

```diff
@@ -82,15 +85,44 @@ func (a *configAgent) recordError(label string) {
 func (a *configAgent) GetConfig(metadata api.Metadata) (api.ReleaseBuildConfiguration, error) {
 	a.lock.RLock()
 	defer a.lock.RUnlock()
-	config, ok := a.configs[metadata.Basename()]
+	config, ok := a.configsByFilename[metadata.Basename()]
 	if !ok {
 		return api.ReleaseBuildConfiguration{}, fmt.Errorf("Could not find config %s", metadata.Basename())
 	}
 	return config, nil
 }
 
+// GetMatchingConfig loads a configuration that matches the metadata,
+// allowing for regex matching on branch names.
+func (a *configAgent) GetMatchingConfig(metadata api.Metadata) (api.ReleaseBuildConfiguration, error) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+	orgConfigs, exist := a.configsByOrgRepo[metadata.Org]
+	if !exist {
+		return api.ReleaseBuildConfiguration{}, fmt.Errorf("could not find any config for org %s", metadata.Org)
+	}
+	repoConfigs, exist := orgConfigs[metadata.Repo]
+	if !exist {
+		return api.ReleaseBuildConfiguration{}, fmt.Errorf("could not find any config for repo %s/%s", metadata.Org, metadata.Repo)
+	}
+	var matchingConfigs []api.ReleaseBuildConfiguration
+	for _, config := range repoConfigs {
+		if regexp.MustCompile(config.Metadata.Branch).MatchString(metadata.Branch) && config.Metadata.Variant == metadata.Variant {
```

Comment:
Do you want it at load-time? Right now it's at the largest scope that the branch exists

Reply:
Just in the very beginning of this func
And also, do not use `MustCompile`, it might panic and that means that a single defunct regex here can kill everything that uses the `configAgent`

Reply:
Err, how do you mean at the "top of this func" ... we are compiling the branches on the config, not the branch that gets passed in. This is the largest scope in which any given branch exists.